### PR TITLE
add test cases related to matview with adhoc ddl change

### PR DESCRIPTION
**Pull request by jxchinaren**

### CODE REVIEW

File: tests/frontend/org/voltdb/TestAdhocAlterTable.java [hunk 1]
```diff
@@ -1122,4 +1123,157 @@ public void testAlterTableENG7304ENG7305() throws Exception
...
+
+            //
+            // DROP a column not in the view succeed (the last column dropped)
+            //
+            // -- JSON plans of the view query will not change
+            checkAlterTableSucceed("alter table FOO drop column NUM2;");
+            assertFalse(doesColumnExist("FOO", "NUM2"));
+
+            //
+            // DROP a column not in the view failed (the last column dropped)
```
Comment (paulmartel) [inline tests/frontend/org/voltdb/TestAdhocAlterTable.java]: comment says "failed" code says "succeed"

File: tests/frontend/org/voltdb/TestAdhocAlterTable.java [hunk 2]
```diff
@@ -1122,4 +1123,157 @@ public void testAlterTableENG7304ENG7305() throws Exception
...
+            assertFalse(doesColumnExist("BAR", "NUM2"));
+
+            //
+            // Rename the source table failed
+            //
+            checkAlterTableFailed("alter table FOO rename to FOO1;", "AdHoc DDL ALTER/RENAME is not yet supported");
+            assertTrue(verifyTableColumnType("FOO", "VAL", "INTEGER"));
+
+            //
+            // Add table to make sure the view is correct
```
Comment (paulmartel) [inline tests/frontend/org/voltdb/TestAdhocAlterTable.java]: Add table? You mean "Add rows to table"?

File: tests/frontend/org/voltdb/TestAdhocAlterTable.java [hunk 3]
```diff
@@ -1122,4 +1123,157 @@ public void testAlterTableENG7304ENG7305() throws Exception
...
+            vt = m_client.callProcedure("@AdHoc", "select val, total from fooview order by 1, 2;").getResults()[0];
+            RegressionSuite.validateTableOfLongs(vt, new long[][]{{-2, 1}, {1, 2}, {2, 1}});
+
+            m_client.callProcedure("@AdHoc", "insert into BAR values(1, 1, 1);");
+            m_client.callProcedure("@AdHoc", "insert into BAR values(1, 1, 2);");
+            m_client.callProcedure("@AdHoc", "insert into BAR values(1, 2, 3);");
+            m_client.callProcedure("@AdHoc", "insert into BAR values(1, -2, 4);");
+
+            vt = m_client.callProcedure("@AdHoc", "select val, total from barview order by 1, 2;").getResults()[0];
+            RegressionSuite.validateTableOfLongs(vt, new long[][]{{1,2}, {2,1}});
```
Comment (paulmartel) [inline tests/frontend/org/voltdb/TestAdhocAlterTable.java]: Add comment explaining why there is no "{-2, 1}" row -- because of the predicate (I HOPE!).

**Comment by paulmartel** [comment]
End of review round.

File: tests/frontend/org/voltdb/TestAdhocAlterTable.java (hunk 2 above)
Comment (jxchinaren) [inline tests/frontend/org/voltdb/TestAdhocAlterTable.java]: yes, add data to the table to verify the view working properly.

**Comment by jxchinaren** [comment]
@paulmartel I added a new commit to resolve your comments, thank you.

**Pull request merged**